src/ape/managers/query.py

```diff
@@ -53,6 +22,48 @@ class QueryManager:
         chain.blocks.query()
     """
 
-    def query(self, query: "Query") -> pd.DataFrame:
-        """ """
-        pass
+    plugin_manager: PluginManager
+    """A reference to the global plugin manager."""
+
+    @cached_property
+    def query_engines(self) -> List[QueryAPI]:
+        return [engine for _, engine in self.plugin_manager.query_engines]
+
+    def __getitem__(self, engine_name: str) -> QueryAPI:
+        """
+        Get an engine by name.
+
+        Args:
+            engine_name (str): The name of the engine to get.
+
+        Returns:
+            :class:`~ape.api.query.QueryAPI`
+        """
+        if engine_name not in self.plugin_manager.query_engines:
+            raise QueryEngineException(engine_name)
```

Comment:
With the above being a dict, you shouldn't have to do this anymore

However, not sure this manager should be user-facing, so I don't think we even need this method for easy access